tests/functional/tests.py

```diff
@@ -628,12 +628,23 @@ def test_negative_put_object_with_path_segment(client, log_output):
                           object_name,
                           io.BytesIO(b''), 0)
     except ResponseError as err:
-        pass
+        if err.code != 'XMinioInvalidObjectName':
+            raise Exception(err)
+        else:
+            pass
     except Exception as err:
         raise Exception(err)
     finally:
         try:
-            client.remove_object(bucket_name, object_name)
+            try:
```

Comment:
There are 2 `try` blocks back to back.
First `try` block is unnecessary.

Reply:
you need it... otherwise, we will crash. Look at the original issue why this fix is necessary. #5954